zip-0227.rst

```diff
@@ -1,8 +1,436 @@
+
+Issuance Key Derivation
+```````````````````````
+
+The issuance master key is generated by choosing a bit sequence uniformly at random from :math:`\mathbb{B}^{\mathbb{Y}[32]}`, like the Orchard spending key [#protocol-orchardkeycomponents]_.
+In :math:`\mathsf{zcashd}`, this key is derived in a manner similar to the generation of the Orchard master key in ZIP 32 [#zip-0032-orchard-master]_, as detailed below:
+
+- Details TBD
+
+The issuance authorizing key and issuance validating key are derived from the issuance master key in an analogous manner to the derivation of the Orchard spend authorizing key and Orchard spend validating key from the Orchard spending key [#protocol-orchardkeycomponents]_, as described below.
+
+- The issuance authorizing key is derived from the issuance master key, :math:`\mathsf{sk}_{\mathsf{iss}}`, as a private signature key:
+
+.. math:: \mathsf{isk} := \mathsf{ToScalar}^{\mathsf{Orchard}}(︀ \mathsf{sk}_{\mathsf{iss}} )
+
+- The issuance validating key is derived from the issuance authorizing key, :math:`\mathsf{isk}`, as a public verification key:
+
+.. math:: \mathsf{ik} := \mathsf{IssueAuthSig}.\mathsf{DerivePublic}(\mathsf{isk})
+
+This allows the issuer to use the same wallet it usually uses to transfer Assets, while keeping a disconnect from the other keys. Specifically, this method is aligned with the requirements and motivation of ZIP 32 [#zip-0032]_. It provides further anonymity and the ability to delegate issuance of an Asset (or in the future, generate a multi-signature protocol) while the rest of the keys remain in the wallet safe.
+
+The derivation of these keys are shown in the following diagram:
+
+.. figure:: zip-0227-key-components-zsa.png
+    :width: 450px
+    :align: center
+    :figclass: align-center
+
+    Diagram of Issuance Key Components for the ZSA Protocol
+
+
+Asset Identifier
+----------------
+
+For every new Asset, there must be a new and unique Asset Identifier. We define this to be a globally unique pair :math:`(\mathsf{ik}, \mathsf{asset\_desc})`, where :math:`\mathsf{ik}` is the issuance key and :math:`\mathsf{asset\_desc}` is a byte string.
+
+A given Asset Identifier is used across all Zcash protocols that support ZSAs -- that is, the Orchard-based ZSA protocol and potentially future Zcash shielded protocols. For this Asset Identifier, we derive an Asset Digest, :math:`\mathsf{AssetDigest}`, which is simply is a :math:`\textsf{BLAKE2b-512}` hash of the Asset Identifier.
+From the Asset Digest, we derive a specific Asset Base within each such shielded protocol (for example :math:`\mathsf{AssetBase}^{\mathsf{Orchard}}_{\mathsf{AssetId}}` for the Orchard-based ZSA protocol), using the applicable hash-to-curve algorithm. This Asset Base is included in shielded notes.
+
+Let 
+
+- :math:`\mathsf{asset\_desc}` be the asset description, a UTF-8 encoded string of up to 512 bytes, which includes any information pertaining to the issuance.
```

Comment:
There is not currently any case in which Zcash consensus depends on the definition of UTF-8. (For memo fields in decrypted shielded coinbase outputs, the plaintext memo is not required to be well-formed UTF-8, since the relevant consensus rule only requires that decryption with the zero $\mathsf{ovk}$ does not return $\bot$.)

Also, "UTF-8 encoded string" is ambiguous as to whether it is required to be well-formed and the applicable version of Unicode (the definition of "well-formed" UTF-8 changed in Unicode 3.1 and [again in Unicode 3.2](https://www.unicode.org/reports/tr28/tr28-3.html#conformance), for example).

Either:
1. Define the asset description as a byte sequence and say that it SHOULD be a well-formed UTF-8 code unit sequence according to Unicode 15.0.0 or later; or
2. Say that it is a consensus rule that the asset description MUST be well-formed UTF-8 code unit sequence according to exactly [Unicode 15.0.0 chapter 3, clause D92](https://www.unicode.org/versions/Unicode15.0.0/ch03.pdf).

In either case, clarify that it MAY include encoded characters that are unassigned in any particular version of Unicode. I suggest 1. rather than 2.

Reply:
Went ahead with alternative 1.